lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2982,11 +3032,32 @@ calculateFeePercentiles
         ErrSelectAssetsSelectionError
             ( SelectionBalanceErrorOf
                 ( UnableToConstructChange
-                    ( UnableToConstructChangeError {requiredCost})
+                    UnableToConstructChangeError{requiredCost}
                 )
             ) -> pure $ Fee requiredCost
         e -> throwE e
 
+-- | Make a pair of fee estimation percentiles more imprecise.
+padFeePercentiles
+    :: ProtocolParameters
+    -> Quantity "byte" Word
+    -- ^ Number of bytes by which to extend the interval in both directions.
+    -> (Percentile 10 Fee, Percentile 90 Fee)
+    -> (Percentile 10 Fee, Percentile 90 Fee)
+padFeePercentiles
+    pp
+    (Quantity byteDelta)
+    (Percentile (Fee a), Percentile (Fee b)) =
+        ( Percentile $ Fee $ a `Coin.difference` coinDelta
+        , Percentile $ Fee $ b `Coin.add` coinDelta
+        )
+  where
+    coinDelta :: Coin
+    coinDelta =
+        Coin.fromNatural . ceiling $ fromIntegral byteDelta * slope feeFunction
+
+    LinearFee feeFunction = pp ^. #txParameters . #getFeePolicy
```

Comment:
Suggestion: explicitly specify the source and target type for `fromIntegral`.
Justification: `fromIntegral` is an unsafe function. If we do use it, then adding types can help readers to judge whether the usage is correct.

```suggestion
      coinDelta :: Coin
      coinDelta
          = Coin.fromNatural
          . ceiling
          $ fromIntegral @Word @Double byteDelta * slope

      LinearFee (LinearFunction {slope}) = pp ^. #txParameters . #getFeePolicy
```